File: content/03/code/java/05--step05-allow-standing-or-located-as-alternates/shouty/src/test/java/shouty/StepDefinitions.java
package shouty;

import io.cucumber.java.PendingException;
import io.cucumber.java.en.Given;
import io.cucumber.java.en.Then;
import io.cucumber.java.en.When;

import static java.util.Arrays.asList;
import static org.junit.Assert.assertEquals;

public class StepDefinitions {
    Person lucy = new Person();
    Person sean = new Person();
    private String messageFromSean;

    @Given("Lucy is located/standing {int} metre(s) from Sean")
    public void lucy_is_metres_from_Sean(Integer distance) {
        throw new PendingException(String.format("Lucy is %d centimetres from Sean", distance * 100));
    }

    @When("Sean shouts {string}")
    public void sean_shouts(String message) {
        sean.shout(message);
        messageFromSean = message;
     }

    @Then("Lucy should hear Sean's message")
    public void lucy_should_hear_Sean_s_message() {
        assertEquals(asList(messageFromSean), lucy.getMessagesHeard());
    }
}

File: content/03/code/dotnet/05--step05-allow-standing-or-located-as-alternates/Shouty/Shouty.Specs/StepDefinitions/StepDefinitions.cs
using System;
using TechTalk.SpecFlow;
using Xunit;

namespace Shouty.Specs.StepDefinitions
{
    [Binding]
    public class StepDefinitions
    {
        private Person lucy = new Person();
        private Person sean = new Person();
        private string messageFromSean;

        [Given("Lucy is located/standing {int} metre(s) from Sean")]
        public void GivenLucyIsLocatedMetresFromSean(int distance)
        {
            throw new NotImplementedException($"Lucy is {distance * 100} centimetres from Sean");
        }

        [When("Sean shouts {string}")]
        public void WhenSeanShouts(string message)
        {
            sean.Shout(message);
            messageFromSean = message;
        }

        [Then("Lucy should hear Sean's message")]
        public void ThenLucyShouldHearSeansMessage()
        {
            Assert.Contains(messageFromSean, lucy.GetMessagesHeard());
        }
    }
}
